joblib/hashing.py

```diff
@@ -146,7 +173,15 @@ def __init__(self, hash_name='md5', coerce_mmap=False):
         else:
             self._getbuffer = memoryview
 
-    def save(self, obj):
+    def persistent_id(self, obj):
+        # Separate dtypes shared by objects in the pickle graph,
+        # to avoid cache invalidation when unpickling a subobject of the graph
+        if isinstance(obj, self.np.dtype):
+            return obj.str
```

Comment:
```
%timeit dumps(a.dtype)
100000 loops, best of 3: 12.5 µs per loop
```

I don't think this is a problem.

However I don't understand the goal of this method. Where is it used?